src/js/components/FormSelect/FormSelect.js

```diff
@@ -1,30 +1,112 @@
 // @flow
 import * as React from 'react';
 import classNames from 'classnames';
+import CreatableSelect from 'react-select/lib/Creatable';
 
 type Props = {
   children: React.ChildrenArray<React.Element<'option'>>,
   className?: string,
+  isClearable?: boolean,
   isInvalid?: boolean,
+  isMulti?: boolean,
   isValid?: boolean,
+  options: { [string]: any },
   size?: 'sm' | 'md' | 'lg',
 };
 
-const FormSelect = ({ children, className, isInvalid, isValid, size, ...other }: Props) => {
+const FormSelect = ({
+  options,
+  className,
+  isInvalid,
+  isValid,
+  size,
+  isMulti,
+  isClearable,
+  ...other
+}: Props) => {
   const classes = classNames(
     className,
-    'form-control',
-    size ? `form-control-${size}` : false,
     {
       'is-invalid': isInvalid,
     },
     { 'is-valid': isValid }
   );
 
+  const getSize = () => {
+    if (size === 'lg') {
+      return {
+        padding: '.5rem 1rem',
+        fontSize: '1.25rem',
+        height: 'inherit',
+      };
+    }
+    if (size === 'sm') {
+      return {
+        padding: '0 .5rem',
+        fontSize: '.875rem',
+        height: 'inherit',
+      };
+    }
+    return {
+      padding: '0.9375rem',
+      fontSize: 'inherit',
+      height: 'calc(2.875rem + 2px)',
+    };
+  };
+
+  const getBorder = (isFocused, isHovered) => {
+    if (isInvalid) return '#f25270';
```

Comment:
`isHovered` and `isFocused` are provided as arguments but `isInvalid` is not.  I think that this function and `getSize` should probably be moved outside `FormSelect` and have all their values provided as arguments - this will make it clearer how they work and, as suggested yesterday, could be tested in isolation.